alexandria/core/models.py

```diff
@@ -172,7 +173,10 @@ def clone(self):
         self.pk = None
         self.save()
 
-        storage = File.content.field.storage
+        storage_backend = settings.ALEXANDRIA_FILE_STORAGE
+        if settings.ALEXANDRIA_ENABLE_AT_REST_ENCRYPTION:
+            storage_backend = "alexandria.storages.backends.s3.SsecGlobalS3Storage"
```

Comment:
From what I understand this is the only place in your changes where storage selection is changed compared to before, right? This may however cause failure because the `DynamicStorageFieldFile`  relies on the `Field.encryption_status` to determine which `storage` to use. The `clone` method should do that as well. If it doesn't and the `settings.ALEXANDRIA_ENABLE_AT_REST_ENCRYPTION=True` will make the copy procedure fail for a `File` instance with `encryption_status` empty and content unencrypted, do you agree or am I getting this wrong?

Reply:
Yes, the previous implementation was completely wrong and with moving it into the field class it also respects the `encryption_status`